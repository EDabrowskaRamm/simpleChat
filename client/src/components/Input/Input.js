import React from 'react'

import { RootForm, StyledInput } from './Input.shards'

const Input = ({
  handleSubmit,
  id,
  value,
  placeholder,
  handleChange
}) => <RootForm onSubmit={handleSubmit}>
    <StyledInput
      type='text'
      id={id}
      value={value}
      placeholder={placeholder}
      onChange={handleChange}
    />
  </RootForm>

export default Input