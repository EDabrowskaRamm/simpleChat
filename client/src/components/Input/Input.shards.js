import styled from '@emotion/styled'

export const RootForm = styled.form`
  width: 100%;
  padding: 15px 0;
`

export const StyledInput = styled.input`
  width: 100%;
  padding: 27px 33px;
  letter-spacing: 2px;
  border: ${({ theme }) => theme.commons.borders.input};
  border-radius: ${({ theme }) => theme.commons.borderRadius.main};

  &:placeholder {
    color: ${({ theme }) => theme.colors.text.placeholder};
  }
`
