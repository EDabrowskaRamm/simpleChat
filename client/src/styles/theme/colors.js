const _white = '#FFF'
const _colorGrey56 = '#868c97'
const _colorGrey61 = '#979c9f'
const _colorGrey63 = '#9a9fa8'
const _colorGrey89 = '#e1e3e5'
const _colorGrey92 = '#ebebeb'

const _pictonBlue = '#3d90f1'
const _oxfordBlue = '#354052'

export const background = {
  primary: _white,
  secondary: _colorGrey92,
}

export const text = {
  primary: _oxfordBlue,
  secondary: _colorGrey63,
  link: _pictonBlue,
  placeholder: _colorGrey61,
  left: _colorGrey56,
}

export const border = {
  input: _colorGrey56,
  chat: _colorGrey89,
  transparent: 'transparent',
}
