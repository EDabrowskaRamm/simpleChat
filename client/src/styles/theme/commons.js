import { border } from './colors'

export const borders = {
  main: `1px solid ${border.chat}`,
  input: `1px solid ${border.input}`,
}
