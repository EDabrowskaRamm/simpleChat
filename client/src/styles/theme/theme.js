import * as colors from './colors'
import * as commons from './commons'

export default {
  colors,
  commons
}
