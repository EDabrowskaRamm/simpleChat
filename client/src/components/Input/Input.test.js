import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'

import Input from './Input'

import theme from '../../styles/theme/theme'

expect.addSnapshotSerializer(serializer)

describe('renders Input', () => {
  it('Snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Input
          handleSubmit={() => { }}
          id={0}
          value='text'
          placeholder='Type in...'
          handleChange={() => { }}
        />
      </ThemeProvider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
