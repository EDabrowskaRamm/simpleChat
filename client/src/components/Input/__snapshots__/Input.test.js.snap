// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input Snapshot 1`] = `
.emotion-1 {
  width: 100%;
  padding: 15px 0;
}

.emotion-0 {
  width: 100%;
  padding: 27px 33px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  border: 2px solid #868c97;
  border-radius: 9px;
}

.emotion-0:placeholder {
  color: #979c9f;
}

<form
  className="emotion-1"
  onSubmit={[Function]}
>
  <input
    className="emotion-0"
    id={0}
    onChange={[Function]}
    placeholder="Type in..."
    type="text"
    value="text"
  />
</form>
`;
